tensorizer/serialization.py

```diff
@@ -56,12 +56,15 @@
 from tensorizer._internal_utils import Chunked as _Chunked
 from tensorizer._internal_utils import _variable_read
 from tensorizer._NumpyTensor import _NumpyTensor
-from tensorizer.stream_io import CURLStreamFile
 
-if torch.cuda.is_available():
-    cudart = torch.cuda.cudart()
-else:
-    cudart = None
+@dataclasses.dataclass
+class PerfStats:
+    file_readinto_millisecs: AtomicUint = dataclasses.field(default_factory=lambda: AtomicUint(width=8))
+    file_readinto_bytes: AtomicUint = dataclasses.field(default_factory=lambda: AtomicUint(width=8))
+    cuda_to_device_millisecs: AtomicUint = dataclasses.field(default_factory=lambda: AtomicUint(width=8))
+    cuda_bytes: AtomicUint = dataclasses.field(default_factory=lambda: AtomicUint(width=8))
+
+perf_stats = PerfStats()
```

Comment:
I'm fine with either. It's mostly for performance measuring of the tensorizer code itself, so maybe internal is appropriate.

Reply:
Internal sounds good for now (so we should go ahead and prefix all of their identifiers with an underscore)
If we expose it we may want to so per-deserializer rather than as global stats, since global stats would probably get confusing in some cases